lib/common/digest.c

```diff
@@ -112,12 +112,13 @@ calculate_xml_digest_v2(xmlNode * source, gboolean do_filter)
          */
 
     } else {
```

Comment:
The `if` block concerns me. If we enter the `if (do_filter && BEST_EFFORT_STATUS)` block, then we're guaranteed to assert since `buffer` will be null. I didn't change it yet because it was already like that.

Reply:
BEST_EFFORT_STATUS is 0. We can just drop that code